quickwit-indexing/src/actors/indexing_server.rs

```diff
@@ -0,0 +1,250 @@
+// Copyright (C) 2021 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::collections::{BTreeMap, HashMap};
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use anyhow::Context;
+use async_trait::async_trait;
+use quickwit_actors::{
+    Actor, ActorContext, ActorExitStatus, ActorHandle, AsyncActor, Mailbox, Observation, Universe,
+};
+use quickwit_config::{IndexerConfig, SourceConfig};
+use quickwit_metastore::Metastore;
+use quickwit_storage::{quickwit_storage_uri_resolver, StorageUriResolver};
+use tokio::sync::oneshot;
+
+use crate::{IndexingPipeline, IndexingPipelineParams, IndexingStatistics};
+
+pub struct IndexingServerClient {
+    universe: Universe,
+    mailbox: Mailbox<IndexingServerMessage>,
+    handle: ActorHandle<IndexingServer>,
+}
+
+impl IndexingServerClient {
+    pub async fn spawn_pipeline(
+        &self,
+        index_id: String,
+        source_opt: Option<SourceConfig>,
+    ) -> anyhow::Result<()> {
+        let (sender, receiver) = oneshot::channel();
```

Comment:
The request/response pattern is not supported by the actor framework so I'm relying on one-shot channels.